### PR TITLE
kernel/binary_manager: print error messages instead of assert when kernel or user binary is not found

### PR DESCRIPTION
**Pull request by jeongarmy**
It is difficult for user to debug it with an assert when kernel or user binary is not found.
So print error messages instead of assert to let the user know why.

- Logs on failure:
binary_manager: =============== !!ERROR!! ==============
binary_manager: ERROR!! Not found valid user binaries.
binary_manager: Please check user binary and smartfs configurations.
binary_manager: ERROR!! Not found kernel partitions.
binary_manager: Please check kernel partition configurations.
binary_manager: Enable CONFIG_DEBUG_BINMGR_ERROR if you want to see the logs for failures.

### CODE REVIEW

**Review by sangam-swami — APPROVED** [review]
looks fine

File: os/kernel/binary_manager/binary_manager.c [hunk 1]
```diff
@@ -197,4 +205,20 @@ int binary_manager(int argc, char *argv[])
...
+	while (1) {
+		lldbg("=============== !!ERROR!! ============== \n");
+#ifdef CONFIG_APP_BINARY_SEPARATION
+		if (!is_found_ubin) {
+			lldbg("ERROR!! Not found valid user binaries.\n");
+			lldbg("Please check user binary and smartfs configurations.\n");
+		}
+#endif
+		if (!is_found_kbin) {
+			lldbg("ERROR!! Not found kernel partitions.\n");
```
Comment (jeongchanKim) [inline os/kernel/binary_manager/binary_manager.c:218]: `binary_manager` is also running on kernel, so I think this message can make confusion.
Please modify the message with more detail.

File: os/kernel/binary_manager/binary_manager.c [hunk 2]
```diff
@@ -197,4 +205,20 @@ int binary_manager(int argc, char *argv[])
...
+errout_with_nobinary:
+	while (1) {
+		lldbg("=============== !!ERROR!! ============== \n");
+#ifdef CONFIG_APP_BINARY_SEPARATION
+		if (!is_found_ubin) {
+			lldbg("ERROR!! Not found valid user binaries.\n");
+			lldbg("Please check user binary and smartfs configurations.\n");
+		}
+#endif
+		if (!is_found_kbin) {
```
Comment (jeongchanKim) [inline os/kernel/binary_manager/binary_manager.c:217]: Is this checking the binary? or partition?
The variable name is binary, but the log message is partition.
Please unify one of them.

File: os/kernel/binary_manager/binary_manager.c [hunk 3]
```diff
@@ -197,4 +205,20 @@ int binary_manager(int argc, char *argv[])
...
+			lldbg("ERROR!! Not found kernel partitions.\n");
+			lldbg("Please check kernel partition configurations.\n");
+		}
+		lldbg("Enable CONFIG_DEBUG_BINMGR_ERROR if you want to see the logs for failures.\n\n");
```
Comment (jeongchanKim) [inline os/kernel/binary_manager/binary_manager.c:221]: In the above, it says that there is no valid binary or partition. That is the log for failure.
So I think `if you want to see the logs for failures` can be changed to include more details.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**